fmriprep/workflows/bold/base.py

```diff
@@ -880,7 +880,10 @@ def _get_series_len(bold_fname):
     if len(img.shape) < 4:
         return 1
 
-    skip_vols = _get_vols_to_discard(img)
+    if isinstance(config.workflow.dummy_scans, int):
+        skip_vols = config.workflow.dummy_scans
+    else:
+        skip_vols = _get_vols_to_discard(img)
```

Comment:
The `config` object is a hulking beast, and I'd prefer to keep its tentacles confined to `init_*_wf()` functions, and pass arguments to standalone functions, to make them easier to reason about.

```Python
def _get_series_len(bold_fname, skip_vols):
    from niworkflows.interfaces.registration import _get_vols_to_discard
    img = nb.load(bold_fname)
    if len(img.shape) < 4:
        return 1

    if skip_vols is None:
        skip_vols = _get_vols_to_discard(img)

    return img.shape[3] - skip_vols
```

And then, above, pass `config.workflow.dummy_scans` to `_get_series_len()`.

Reply:
We probably want to make an overhaul of the documentation, on the code guidelines section) and:
- mention this (please do not use config deep in the workflow)
- recommend black for everything except the `workflow.connect` function

Reply:
I haven't actually looked at the code guidelines in a while, so I read them again just now. 

I realized that I probably also should have considered the branch naming/pull request naming, and added `[ENH]`.